test/treeTrimmingTest.js

```diff
@@ -0,0 +1,347 @@
+            "children": [
+                {
+                    "id": "a",
+                    "cat": "x0"
+                },
+                {
+                    "id": "b",
+                    "cat": "x0"
+                }
+            ]
+        },
+        {
+            "cat": "xp",
+            "id": "XP_4",
+            "children": [
+                {
+                    "id": "c",
+                    "cat": "x0",
+                    "silent": true
+                }
+            ]
+        }
+    ]
+};
+
+stree7 = {
+    "id": "T7",
+    "cat": "cp",
+    "children": [
+        {
+            "cat": "xp",
+            "id": "XP_5",
+            "children": [
+                {
+                    "id": "a",
+                    "cat": "x0"
+                },
+                {
+                    "id": "b",
+                    "cat": "x0"
+                }
+            ]
+        },
+        {
+            "cat": "xp",
+            "id": "XP_4",
+            "children": [
+                {
+                    "id": "c",
+                    "cat": "monero"
+                }
+            ]
+        }
+    ]
+};
+
+stree8 = {
+    "id": "T8",
+    "cat": "cp",
+    "children": [
+        {
+            "cat": "xp",
+            "id": "XP_6",
+            "children": [
+                {
+                    "id": "a",
+                    "cat": "x0"
+                },
+                {
+                    "id": "b",
+                    "cat": "x0"
+                }
+            ]
+        },
+        {
+            "cat": "xp",
+            "id": "XP_5",
+            "children": [
+                {
+                    "id": "c",
+                    "cat": "x0",
+                    "silent": true
+                },
+                {
+                    "id": "d",
+                    "cat": "dogecoin"
+                }
+            ]
+        }
+    ]
+};
+
+stree9 = {
+    "id": "T9",
+    "cat": "cp",
+    "children": [
+        {
+            "cat": "xp",
+            "id": "XP_5",
+            "children": [
+                {
+                    "cat": "cp",
+                    "id": "XP_4",
+                    "children": [
+                        {
+                            "id": "a",
+                            "cat": "x0",
+                            "silent": true
+                        },
+                        {
+                            "cat": "xp",
+                            "id": "XP_3",
+                            "children": [
+                                {
+                                    "id": "b",
+                                    "cat": "x0"
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        }
+    ]
+};
+
+function treeTrimmingTest(){
+    describe('Tree Trimming Tests', function() {
+        it('Does nothing if tree is OK', function() {
```

Comment:
It would be helpful if these trees were defined inside the it() statements so that it was easy to see the code for them, or if the text of the it() statement included a call to parenthesizeTree(stree1) (or whatever the relevant tree was) so that someone looking at the test can tell easily what the input was.

Reply:
Also as it stands, these tests are all dependent on the implementation of parenthesizeTree(). Instead of comparing the bracketed strings, change them to compare JSON objects, using JSON.stringify() on the output of removeSpecifiedNodes(). That way the tests won't all break unnecessarily the next time parenthesizeTree() gets adjusted.

Reply:
Trees now show when the it statement is clicked.
JSON.stringify() is now used to check instead of parenthesizeTrees().